File: source/toy/file/loader/Png.cpp
#include <cstdlib>
#include "toy/ImageBuffer.hpp"
#include "toy/file/loader/Png.hpp"
#include "toy/file/loader/Image.hpp"

namespace toy{ namespace file{ namespace loader{
extern bool _SavePng(const std::string&,toy::ImageBuffer *);
}}}

namespace toy{
namespace file{

bool loader::png::Load(toy::File *pIO,toy::ImageBuffer *map)
{
	return loader::Load(pIO,map);
}

bool loader::png::Save(std::string filename,toy::ImageBuffer *map)
{
	return loader::_SavePng(filename,map);
}

}}

File: examples/usage/ToyBox/modules/toy/file/loader/png.lua

local  bug = require "toy.debug"
local  fs  = require "toy.file"
local  png = require "toy.file.loader.png"

local function create_window()
	local  config =
	{
		title = "Show a png image",
		width = 800,
		height = 600,
	}

	return require("toy.graph").create_window(config)
end

local function load_image_file(folder,filename)
	local  stream = fs.create_stream("default")
	local  buffer = fs.create_image_buffer()

	stream:open_dir(folder)
	stream:open(filename)
	png.load_image(stream,buffer)     -- Load data from stream, then put into the buffer.

	return buffer
end

local function main()
	local  window = create_window()
	local  image  = window:create_scene():new_image(load_image_file("D:/projects/ToyBox/resource","002.png"))

	image:visible(true)
	image:set_position(-400,-300)

	while window:handle_event() do
		window:display()
	end
end

if bug.catch(main) then
	bug.oops()
end
